src/Altinn.Platform/Altinn.Platform.Events/Events/Controllers/SubscriptionController.cs

```diff
@@ -0,0 +1,265 @@
+using System.Threading.Tasks;
+using Altinn.Common.AccessToken.Configuration;
+using Altinn.Common.PEP.Interfaces;
+using Altinn.Platform.Events.Authorization;
+using Altinn.Platform.Events.Configuration;
+using Altinn.Platform.Events.Models;
+using Altinn.Platform.Events.Services.Interfaces;
+using Altinn.Platorm.Events.Extensions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Events.Controllers
+{
+    /// <summary>
+    /// Controller to handle administration of event subscriptions
+    /// </summary>
+    [Route("events/api/v1/subscription")]
+    [ApiController]
+    public class SubscriptionController : ControllerBase
+    {
+        private readonly ISubscriptionService _eventsSubscriptionService;
+        private readonly IRegisterService _registerService;
+        private readonly ILogger _logger;
+        private readonly string _eventsBaseUri;
+        private readonly AuthorizationHelper _authorizationHelper;
+        private readonly AccessTokenSettings _accessTokenSettings;
+
+        private const string OrganizationPrefix = "/organization/";
+        private const string PersonPrefix = "/person/";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SubscriptionController"/> class.
+        /// </summary>
+        public SubscriptionController(
+            ISubscriptionService eventsSubscriptionService,
+            IRegisterService registerService,
+            IOptions<GeneralSettings> settings,
+            ILogger<EventsController> logger,
+            IPDP pdp,
+            IOptions<AccessTokenSettings> accessTokenSettings)
+        {
+            _registerService = registerService;
+            _logger = logger;
+            _eventsSubscriptionService = eventsSubscriptionService;
+            _eventsBaseUri = $"https://platform.{settings.Value.Hostname}";
+            _authorizationHelper = new AuthorizationHelper(pdp);
+            _accessTokenSettings = accessTokenSettings.Value;
+        }
+
+        /// <summary>
+        /// Method to register an event
+        /// </summary>
+        /// <param name="eventsSubscription">The subscription details</param>
+        /// <returns></returns>
+        [HttpPost()]
```

Comment:
( ) can be removed if there is no additional routing.

Reply:
Fixed